### ISSUE TITLE
VoiceOver skips legends after following in-page link on iOS 10

### ISSUE THREAD

**Issue by selfthinker** [alphagov/reported-bugs#37]
**Upstream bug:** https://bugs.webkit.org/show_bug.cgi?id=179012
**Date:** 2017-10-30
**Reported by:** Anika Henke
**Related to:** n/a

---

### Overview

When following in-page links, VoiceOver doesn't always read out the next text. This is at least true for paragraphs, inputs and legends. This is not too bad because the text gets read out when swiping further, except for legends. When jumping to a legend and swiping further, the text of the legend is entirely skipped, even when swiping back and forth again. This does not happen when swiping without having followed the link.
It's difficult to tell if this is still true on iOS 11 due to another bug [https://bugs.webkit.org/show_bug.cgi?id=179011].

### Steps to reproduce

1. Activate VoiceOver (on iOS 10)
2. Open http://jsbin.com/wumitiw in Safari
3. Navigate to "link to legend"
4. Follow the link (i.e. double tap)
5. Swipe further

### Actual result

The text for the legend is not read out.

### Expected result

The legend should be read out, if not directly after following the link, at least after swiping further.

### Why is this a problem?

Legends (and fieldsets) are not always just used to group form elements, they can contain information that is vital to understand and complete a form. E.g. with Yes and No questions, you would have Yes and No labels for the radio buttons but the actual question would often be in a legend. It would be impossible for screen reader users to know what the question was if they followed a link to it. And in-page links are typical for pointing users to specific errors after they submitted a form.
